oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/PathElementComparator.java

```diff
@@ -34,7 +35,7 @@ public PathElementComparator() {
     }
 
     public PathElementComparator(Iterable<String> preferredPathElements) {
-        this.preferred = ImmutableSet.copyOf(preferredPathElements);
+        this.preferred = Collections.unmodifiableSet(SetUtils.toLinkedSet(preferredPathElements));
```

Comment:
Here we do not need to preserve insertion order. `Sets.of()` is enough.

Reply:
The issue with `Sets.of` is that, we need to make sure that we don't even pass `null` to `Set.contains()`.

Reply:
In this case, we don't pass `null` (this is indexing code, I am familiar with it).

Reply:
`Set.of()` doesn't accept `Iterable` and `Set.copyOf` only accepts `Collection`.